include/mcmini/misc/cond/MCConditionVariableSingleGroupPolicy.hpp

```diff
@@ -13,6 +13,7 @@ struct ConditionVariableSingleGroupPolicy :
   virtual void receive_broadcast_message() override;
   virtual void wake_thread(tid_t tid) override;
   virtual void add_waiter(tid_t tid) override;
+  virtual bool has_waiters() override;
```

Comment:
This should read now
```cpp
virtual bool has_waiters() const override;
```

Reply:
And we discussed, but forgot to save that at line 45, we need `return ! this->wait_queue.empty();